lazy_tensor_core/lazy_tensor_core/csrc/init_python_bindings.cpp

```diff
@@ -680,14 +680,22 @@ void InitLtcModuleBindings(py::module m) {
           auto post_order = Util::ComputePostOrder(roots);
           std::vector<int64_t> tensor_ids;
           std::vector<at::IValue> ivalues;
+
+          std::unordered_set<BackendData::Handle> data_handles_;
           for (auto nodeptr : post_order) {
             if (nodeptr->op() == *torch::lazy::ltc_device_data) {
               const auto* device_data_node = torch::lazy::NodeCast<torch::lazy::DeviceData>(nodeptr, *torch::lazy::ltc_device_data);
+
               auto infoptr = device_data_node->data()->info();
               auto deviceDataInfoPtr = (torch::lazy::LazyGraphExecutor::DeviceDataInfo*) infoptr;
-              tensor_ids.push_back(deviceDataInfoPtr->tensor_id);
-
               auto* tsDataPtr = (torch_lazy_tensors::compiler::TSData*) device_data_node->data().get();
+
+              // dedup DeviceData by handle
+              auto handle = tsDataPtr->GetHandle();
+              if (!data_handles_.insert(handle).second) {
```

Comment:
Maybe `data_handles_ .count(handle)`?

Reply:
haha yea this is a confusing syntax, I have to look it up to know what .second does for insert :(

Reply:
I just realized that Shunting wanted to combine the insert and the duplication check in one line.

Reply:
Never mind. You can disregard my comment.